// frontend/src/tests/api-test.js
import axios from 'axios';

// 配置基础URL
const api = axios.create({
  baseURL: 'http://localhost:8080',
  timeout: 10000,
  headers: {
    'Content-Type': 'application/json',
  },
});

// 模拟用户ID
const MOCK_USER_ID = 1;

// 测试购物车API
async function testCartAPI() {
  console.log('===== 测试购物车API =====');
  try {
    // 获取购物车列表
    const cartResponse = await api.get(`/api/cart/${MOCK_USER_ID}`);
    console.log('购物车列表:', cartResponse.data);

    // 添加商品到购物车
    const addToCartResponse = await api.post(`/api/cart/${MOCK_USER_ID}/add`, {
      productId: 1,
      quantity: 2
    });
    console.log('添加商品到购物车:', addToCartResponse.data);

    // 更新购物车商品数量
    const updateCartResponse = await api.put(`/api/cart/${MOCK_USER_ID}/update`, {
      cartItemId: 1,
      quantity: 3
    });
    console.log('更新购物车商品数量:', updateCartResponse.data);

    // 获取购物车商品数量
    const cartCountResponse = await api.get(`/api/cart/${MOCK_USER_ID}/count`);
    console.log('购物车商品数量:', cartCountResponse.data);

    console.log('购物车API测试成功');
  } catch (error) {
    console.error('购物车API测试失败:', error.response?.data || error.message);
  }
}

// 测试商品API
async function testProductAPI() {
  console.log('===== 测试商品API =====');
  try {
    // 获取商品列表
    const productsResponse = await api.get('/api/products');
    console.log('商品列表:', productsResponse.data);

    // 获取商品详情
    const productDetailResponse = await api.get('/api/products/1');
    console.log('商品详情:', productDetailResponse.data);

    // 搜索商品
    const searchResponse = await api.get('/api/products/search', {
      params: { keyword: '手机' }
    });
    console.log('搜索商品:', searchResponse.data);

    console.log('商品API测试成功');
  } catch (error) {
    console.error('商品API测试失败:', error.response?.data || error.message);
  }
}

// 测试订单API
async function testOrderAPI() {
  console.log('===== 测试订单API =====');
  try {
    // 获取用户订单列表
    const ordersResponse = await api.get(`/api/orders/user/${MOCK_USER_ID}`);
    console.log('订单列表:', ordersResponse.data);

    // 创建订单
    const createOrderResponse = await api.post(`/api/orders/${MOCK_USER_ID}`, {
      cartItemIds: [1, 2],
      addressId: 1,
      paymentMethod: 'credit_card',
      expectedAmount: 199.99,
      remark: '测试订单'
    });
    console.log('创建订单:', createOrderResponse.data);

    // 获取订单详情
    if (createOrderResponse.data.code === 200) {
      const orderId = createOrderResponse.data.data.orderId;
      const orderDetailResponse = await api.get(`/api/orders/${orderId}`);
      console.log('订单详情:', orderDetailResponse.data);
    }

    console.log('订单API测试成功');
  } catch (error) {
    console.error('订单API测试失败:', error.response?.data || error.message);
  }
}

// 测试支付API
async function testPaymentAPI() {
  console.log('===== 测试支付API =====');
  try {
    // 获取分期方案
    const installmentPlansResponse = await api.get('/api/payments/installment-plans', {
      params: { amount: 1000 }
    });
    console.log('分期方案:', installmentPlansResponse.data);

    // 获取用户信用卡
    const creditCardsResponse = await api.get(`/api/payments/credit-cards/${MOCK_USER_ID}`);
    console.log('用户信用卡:', creditCardsResponse.data);

    console.log('支付API测试成功');
  } catch (error) {
    console.error('支付API测试失败:', error.response?.data || error.message);
  }
}

// 测试用户API
async function testUserAPI() {
  console.log('===== 测试用户API =====');
  try {
    // 获取用户信息
    const userResponse = await api.get(`/api/users/${MOCK_USER_ID}`);
    console.log('用户信息:', userResponse.data);

    // 获取用户地址
    const addressesResponse = await api.get(`/api/users/${MOCK_USER_ID}/addresses`);
    console.log('用户地址:', addressesResponse.data);

    // 获取用户积分
    const pointsResponse = await api.get(`/api/users/${MOCK_USER_ID}/points`);
    console.log('用户积分:', pointsResponse.data);

    console.log('用户API测试成功');
  } catch (error) {
    console.error('用户API测试失败:', error.response?.data || error.message);
  }
}

// 运行所有测试
async function runAllTests() {
  console.log('开始API集成测试...');
  
  await testProductAPI();
  await testCartAPI();
  await testUserAPI();
  await testOrderAPI();
  await testPaymentAPI();
  
  console.log('API集成测试完成');
}

// 执行测试
runAllTests();

export { testProductAPI, testCartAPI, testUserAPI, testOrderAPI, testPaymentAPI, runAllTests };
