// frontend/src/services/cartService.ts
import api from './api';
import { ApiResponse, CartItem, Product } from '../types';

/**
 * 购物车服务类
 */
export class CartService {
  /**
   * 获取用户购物车商品列表
   */
  static async getCartItems(userId: number): Promise<CartItem[]> {
    const response = await api.get<ApiResponse<CartItem[]>>(`/api/cart/${userId}`);
    return response.data.data;
  }

  /**
   * 添加商品到购物车
   */
  static async addToCart(userId: number, productId: number, quantity: number): Promise<string> {
    const response = await api.post<ApiResponse<string>>(`/api/cart/${userId}/add`, {
      productId,
      quantity
    });
    return response.data.message;
  }

  /**
   * 更新购物车商品数量
   */
  static async updateCartItemQuantity(userId: number, cartItemId: number, quantity: number): Promise<string> {
    const response = await api.put<ApiResponse<string>>(`/api/cart/${userId}/update`, {
      cartItemId,
      quantity
    });
    return response.data.message;
  }

  /**
   * 从购物车删除商品
   */
  static async removeFromCart(userId: number, cartItemId: number): Promise<string> {
    const response = await api.delete<ApiResponse<string>>(`/api/cart/${userId}/remove/${cartItemId}`);
    return response.data.message;
  }

  /**
   * 根据商品ID从购物车删除商品
   */
  static async removeFromCartByProductId(userId: number, productId: number): Promise<string> {
    const response = await api.delete<ApiResponse<string>>(`/api/cart/${userId}/remove-product/${productId}`);
    return response.data.message;
  }

  /**
   * 清空用户购物车
   */
  static async clearCart(userId: number): Promise<string> {
    const response = await api.delete<ApiResponse<string>>(`/api/cart/${userId}/clear`);
    return response.data.message;
  }

  /**
   * 获取用户购物车商品数量
   */
  static async getCartItemCount(userId: number): Promise<number> {
    const response = await api.get<ApiResponse<number>>(`/api/cart/${userId}/count`);
    return response.data.data;
  }
}
